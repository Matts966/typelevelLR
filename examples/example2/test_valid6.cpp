
#include <iostream>
#include "example2.hpp"
using namespace example2;

int main() {
  std::cout << *begin()->num( 1 )->mul()->lp()->num( 2 )->add()->num( 3 )->rp()->mul()->num( 4 )->end() << std::endl;
  return 0;
}
