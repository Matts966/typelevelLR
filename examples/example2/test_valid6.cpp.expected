TToE(Mul(Mul(FToT(Num(1)), Paren(Add(TToE(FToT(Num(2))), FToT(Num(3))))), Num(4)))